### ISSUE TITLE
Snap mode not working for landuse=

### ISSUE THREAD

**Issue by jidanni** [MarcusWolschon/osmeditor4android#2267]
With Snap mode on, I tried to plot a barrier=wall using the same corners as an already existing
landuse=commercial .

Alas, if you blow up the screen you'll see Snap mode is broken in this case.

![Screenshot_20230502_224206_Vespucci.jpg](https://user-images.githubusercontent.com/1230959/235702012-e33fb091-5bed-485b-a239-a2cce886a3b5.jpg)

**Comment by jidanni** [comment]
And Join is not offered. The only thing that is offered is  splitting them even more,
![Screenshot_20230502_225234_Vespucci.jpg](https://user-images.githubusercontent.com/1230959/235703868-efa7ed09-d20d-4a3a-a12a-687e6580508d.jpg)

**Comment by jidanni** [comment]
Wait. In fact how you press your finger on such a very small area can give you three different kinds of results.

![Screenshot_20230502_230215_Vespucci.jpg](https://user-images.githubusercontent.com/1230959/235706391-1db22f37-2f72-4c2d-9a76-e661cbea120a.jpg)

![Screenshot_20230502_225824_Vespucci.jpg](https://user-images.githubusercontent.com/1230959/235706433-19fc2d67-44ad-4708-b0fe-596d70d2004b.jpg)

![Screenshot_20230502_225757_Vespucci.jpg](https://user-images.githubusercontent.com/1230959/235706473-2accc69e-2ee5-490e-b893-22a1ad82a0b0.jpg)

So indeed it is attached to this and that and this and that and this...

In other words we're very not capable of doing any kind of microsurgery.

The only way to clear it up would probably be remove all the nodes and put them one way by one back in place.

Seems there needs to be some dedicated microsurgery tools here.

https://www.openstreetmap.org/node/10862537468 et al.

Yes you might say, "Well Dan, you already attached the thing to that and that."
Yes the problem is it's hard to tell.

**Comment by jidanni** [comment]
One thing that might help would be, on each node, have a number,  showing how many things it is a member of.

This might be in some mode the user taps to enable.

**Comment by simonpoole** [comment]
I've got no idea what you are doing. 

For starters "snaping" doesn't depend (at all) on how objects are tagged just on proximity.  2nd you have one node that is a member of both ways (so snaping did work, you simply had two nodes there) and one that is just a member of the landuse. Select one of the two nodes and merge it with the other one to remove it.

**Comment by simonpoole** [comment]
PS: shouldn't you have the large drag area turned on?

**Comment by jidanni** [comment]
Okay, at full zoom I could finally select and merge the node in question.

![Screenshot_20230504-082220_1 - Copy.jpg](https://user-images.githubusercontent.com/1230959/236079229-cbfff23d-8a46-4140-96b6-28610ce5c43b.jpg)
